src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -313,56 +301,47 @@ private void ActOnDocumentInMultipleProjects(string filePath, Action<IProjectSna
         }
     }
 
-    private ProjectKey AddProjectCore(ProjectSnapshotManager.Updater updater, string filePath, string intermediateOutputPath, RazorConfiguration? configuration, string? rootNamespace, string? displayName)
+    private ProjectKey AddProjectCore(ProjectSnapshotManager.Updater updater, HostProject hostProject)
     {
-        var normalizedPath = FilePathNormalizer.Normalize(filePath);
```

Comment:
I love that you spotted this! It was the only place that I was aware of removing a call to `FilePathNormalizer.Normalize(...)`. I _fully_ agree that the whole file-normalization business is super error prone, and I think `HostProject` is great place to unify some of that. I'll take a closer look at that for this pull request.

> Thankfully the Good Ship Cohosting should save us :D

goodshipcohosting.com appears to be available for purchase.

Reply:
FWIW, as I've started looking at this, I'm finding that it's unclear whether `HostProject.FIlePath` should be normalized or not. Currently, it looks like the VS `ProjectSnapshotManager` does not contain normalized project paths but the language server `ProjectSnapshotManager` does. Since CPS only accesses the VS `ProjectSnapshotManager`, I can't see how it would be affected by the language server's `ProjectSnapshotManager` suddenly not having normalized file paths. This looks to me like a mess and I'm not sure that I know how it's supposed to work.

Reply:
The VS side not having normalized paths is not too surprising I guess, since it doesn't have a RazorProjectService, so perhaps my worry about that specific location wasn't valid (or is already broken 😁). I still worry about changing the language server side to not be normalized, but perhaps its worth rolling the die for a preview 1?

On the other hand, I don't think normalizing everywhere is likely to introduce issues, the only downside really is reinforcing the cargo cult.

Reply:
> perhaps its worth rolling the die for a preview 1

That doesn't feel good to me. I want to get to the bottom of this. :smile: